pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+package nsmconnect;
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string Name = 1;
+    string Namespace = 2;
+    Label Labels = 3;
+}
+
+enum InterfaceType {
+    KernelInterface = 0;
+    VhostUser = 1;
+    Memif = 2;
+    Other = 3;
+  }
+
+message MechanismType {
+    InterfaceType type = 1;
+  }
+
+message AffinityPreference {
+    string Predicate = 1;
+}
+
+message ConnectionRequest {
+
+    Metadata Metadata = 1;
+    
+    string NetworkServiceName = 2;
```

Comment:
At some point we probably need to do one of two things:

a)  Add the ability to specify the name of the channel within the NetworkService
or
b)  Decide we aren't doing multiple channels per Network Service.  

When last discussed in the NSM meeting, the concensus was to keep multiple channels per Network Service.  The reasoning was that it might be useful and it was perhaps a bit to early to trim that back, so it definitely can be revisited (ie, decision clearly not final).

Reply:
It is fine to add a channel to the request, but I do have a question, how the client knows about existing channels? Maybe we should add a `Discovery` request, which will return to the client all existing `NetworkService`s and NetworkService_Channel`s. In this case it will make sense to specify NetworkService name and Channel in ConnectionRequest. WDYT?